### ISSUE TITLE
[validation config file ]:  command or other way to validate config file 

### ISSUE THREAD

**Issue by saharmethneni** [prometheus/cloudwatch_exporter#484]
# Command or other way to validate the configuration file
### It will be better to validate the configuration file to detect errors before loading it .

This will allow the user to only load the correct configuration files and not break what is already running

**Comment by matthiasr** [comment]
I agree, this would be a great thing to have. It could be a function of the regular exporter, if supplied with a flag it would check the configuration and exit (status code 0 if correct, 1 if incorrect).

As the base case, I think this should be a purely offline syntax check. However, given the nature of the exporter, it would be useful to have an extended / online check mode that actually connects to CloudWatch to validate that it can fetch the metrics, dimensions are correct, etc. Iit should not stop at the first error but rather list all (possible) problems.
